lib/librtc/librtc.h

```diff
@@ -0,0 +1,159 @@
+		typedef bool librtc_bool;
+	#elif __STDC_VERSION__ >= 199901L
+		typedef _Bool librtc_bool;
+	#elif __GNUC__ >= 15
+		typedef unsigned char __attribute__((hardbool(0, 1))) librtc_bool;
+	#else
+		typedef unsigned char librtc_bool;
+	#endif
+#endif
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/* A structure storing the calendar time
+ * Uses the same format as the tm struct in the C standard library
+ */
+typedef struct {
+	int tm_sec; // seconds after the minute [0-59]
+	int tm_min; // minutes after the hour [0-59]
+	int tm_hour; // hours since midnight [0-23]
+	int tm_mday; // day of the month [1-31] (remember this starts from 1, not 0)
+	int tm_mon; // months since January [0-11]
+	int tm_year; // years since 1900 (remember the epoch is 1900 AD, not 1 BC)
+	int tm_wday; // days since Sunday [0-6]
+	int tm_yday; // days since January 1 [0-365]
+	int tm_isdst; // Field included for the sake of matching the C standard library. Has no effect.
+} librtc_time;
+
+
+/* Initializes the RTC clock if it has not already been initialized.
+ * Returns true if the flashcart or emulator supports a realtime clock.
+ * Note that some emulators and flashcarts may require the RTC bit to be enabled in the Everdrive ROM header.
+ * 
+ * It is not strictly necessary to call librtc_init as it will be automatically initialized when first calling
+ * librtc_get_time or librtc_get_time_raw if not yet initialized. However, to workaround bugs with some flashcarts,
+ * there will be a stutter on console the first time librtc_get_time or librtc_get_time_raw is called if librtc_init
+ * was not already called 20 milliseconds ago or more. See librtc_ready for more information.
+ */
+librtc_bool librtc_init();
+
+/* Returns false if the RTC has not been initialized or if the rom is being played on console and 20 milliseconds have not yet
+ * passed since the RTC was initialized. Otherwise, returns true, even if RTC is not supported by the flashcart or emulator.
+ *
+ * On console, there is a 20 millisecond delay between initializing the RTC and when the time can first be fetched.
+ * This delay exists to wokraround some buggy flashcarts that do not properly set the RTC status. To ensure that the RTC has had
+ * enough time to initialize on these buggy flashcarts, a 20 millisecond delay is enforced to avoid returning garbage data.
+ *
+ * NOTE: The COUNT register on the CPU is used to determine the time passed. Since this counter overflows every ~90 seconds,
+ * if the very first call to librtc_get_time or librtc_get_time occurs precisely within a 20 millisecond window after one of the
+ * ~90 second cycles since initializing the RTC, it will incorrectly believe that 20 milliseconds have not yet passed.
+ * To avoid this, you can call librtc_ready at a regular interval (less then 90 seconds), and the 20 millisecond delay will be
+ * permanently marked as complete once it has passed, ensuring that COUNT overflow will no longer be a concern.
+ */
```

Comment:
Any reason we can't just use `osSetTime` here to deal with this?

Reply:
That doesn't actually set the time on the RTC. In fact, its not related to the RTC at all: https://github.com/n64decomp/libreultra/blob/1aca5c13ca041cef86f8dc194b727361dad9c09b/src/os/settime.c

And even if it did, most emulators don't support that.

Reply:
Plus it wouldn't work when interrupts are disabled.